### PR TITLE
added azure vm plugin vmVTPMEnabled

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

File: plugins/azure/virtualmachines/vmVTPMEnabled.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'VM vTPM Enabled',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensure Virtual Trusted Platform Module (vTPM) is enabled for Azure virtual machines (VM).',
+    more_info: 'vTPM is TPM2.0 compliant and enhances security by validating VM boot integrity and providing a secure storage mechanism for keys and secrets.',
```
Comment (mehakseedat63) [inline plugins/azure/virtualmachines/vmVTPMEnabled.js:9]:
```suggestion
    more_info: 'vTPM is TPM2.0 compliant and enhances security by validating VM boot integrity and providing a secure storage mechanism for keys and secrets.',
```
add a little more info in here

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

File: plugins/azure/virtualmachines/vmVTPMEnabled.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'VM vTPM Enabled',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    description: 'Ensure Virtual Trusted Platform Module (vTPM) is enabled for Azure virtual machines (VM).',
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/vmVTPMEnabled.js:8]:
```suggestion
    description: 'Ensure that Virtual Trusted Platform Module (vTPM) is enabled for Azure virtual machines.',
```

File: plugins/azure/virtualmachines/vmVTPMEnabled.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            }
+
+            if (!virtualMachines.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Machines found', location);
+                return rcb();
+            }
+
+            virtualMachines.data.forEach(virtualMachine => {
+                if (virtualMachine.securityProfile && virtualMachine.securityProfile.uefiSettings && virtualMachine.securityProfile.uefiSettings.vTpmEnabled) {
+                    helpers.addResult(results, 0, 'vTPM is enabled for Azure Virtual Machine', location, virtualMachine.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/vmVTPMEnabled.js:37]:
```suggestion
                    helpers.addResult(results, 0, 'vTPM is enabled for virtual machine', location, virtualMachine.id);
```

File: plugins/azure/virtualmachines/vmVTPMEnabled.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'vTPM is not enabled for Azure Virtual Machine', location, virtualMachine.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/vmVTPMEnabled.js:39]:
```suggestion
                    helpers.addResult(results, 2, 'vTPM is not enabled for virtual machine', location, virtualMachine.id);
```

File: plugins/azure/virtualmachines/vmVTPMEnabled.spec.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                done();
+            });
+        });
+
+        it('should give failing result if vTPM is not enabled', function(done) {
+            const cache = createCache([virtualMachines[1]]);
+            selectVTPM.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('vTPM is not enabled for Azure Virtual Machine');
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/vmVTPMEnabled.spec.js:81]:
```suggestion
                expect(results[0].message).to.include('vTPM is not enabled for virtual machine');
```

File: plugins/azure/virtualmachines/vmVTPMEnabled.spec.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                done();
+            });
+        });
+
+        it('should give passing result if vTPM is enabled', function(done) {
+            const cache = createCache([virtualMachines[0]]);
+            selectVTPM.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('vTPM is enabled for Azure Virtual Machine');
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/vmVTPMEnabled.spec.js:70]:
```suggestion
                expect(results[0].message).to.include('vTPM is enabled for virtual machine');
```

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**